tests/features/records-tags.spec.ts

```diff
@@ -0,0 +1,599 @@
+
+        // negative result same tag different value
+        let tagsQueryNegative = await TestDataGenerator.generateRecordsQuery({
+          author : alice,
+          filter : {
+            tags: {
+              numberTag: 54566974, // off by one
+            }
+          }
+        });
+        let tagsQueryNegativeReply = await dwn.processMessage(alice.did, tagsQueryNegative.message);
+        expect(tagsQueryNegativeReply.status.code).to.equal(200);
+        expect(tagsQueryNegativeReply.entries?.length).to.equal(0);
+
+        // negative result different tag same value
+        tagsQueryNegative = await TestDataGenerator.generateRecordsQuery({
+          author : alice,
+          filter : {
+            tags: {
+              otherTag: 54566975,
+            }
+          }
+        });
+        tagsQueryNegativeReply = await dwn.processMessage(alice.did, tagsQueryNegative.message);
+        expect(tagsQueryNegativeReply.status.code).to.equal(200);
+        expect(tagsQueryNegativeReply.entries?.length).to.equal(0);
+      });
+
+      it('should be able to filter by boolean match', async () => {
+        const alice = await TestDataGenerator.generateDidKeyPersona();
+
+        // write a record with a true boolean value tag
+        const tagsRecordTrue = await TestDataGenerator.generateRecordsWrite({
+          author    : alice,
+          published : true,
+          schema    : 'post',
+          tags      : {
+            booleanTag: true,
+          }
+        });
+
+        const tagsRecordTrueReply = await dwn.processMessage(alice.did, tagsRecordTrue.message, { dataStream: tagsRecordTrue.dataStream });
+        expect(tagsRecordTrueReply.status.code).to.equal(202);
+
+        // write a record with a false boolean value tag
+        const tagsRecordFalse = await TestDataGenerator.generateRecordsWrite({
+          author    : alice,
+          published : true,
+          schema    : 'post',
+          tags      : {
+            booleanTag: false,
+          }
+        });
+
+        const tagsRecordFalseReply = await dwn.processMessage(alice.did, tagsRecordFalse.message, { dataStream: tagsRecordFalse.dataStream });
+        expect(tagsRecordFalseReply.status.code).to.equal(202);
+
+        // do an exact match for the true tag value
+        const tagsQueryMatchTrue = await TestDataGenerator.generateRecordsQuery({
+          author : alice,
+          filter : {
+            tags: {
+              booleanTag: true
+            }
+          }
+        });
+
+        const tagsQueryMatchTrueReply = await dwn.processMessage(alice.did, tagsQueryMatchTrue.message);
+        expect(tagsQueryMatchTrueReply.status.code).to.equal(200);
+        expect(tagsQueryMatchTrueReply.entries?.length).to.equal(1);
+        expect(tagsQueryMatchTrueReply.entries![0].recordId).to.equal(tagsRecordTrue.message.recordId);
+
+        // do an exact match for the true tag value
```

Comment:
Sanity:

```suggestion
        // negative result same tag different value
```

Reply:
Sanity: but it is the same tag (`booleanTag`) though?

Reply:
Whoops, i reworded those comments a bit and added a better scenario description.